src/hooks/usePressDetection.tsx

```diff
@@ -0,0 +1,23 @@
+import { useRef } from 'react'
+
+export const usePressDetection = (): [() => void, () => boolean] => {
```

Comment:
Needs test

Reply:
Tests addressed. New provider created in order to make testing easier. `useDragGesture` hook was taken as example for that.